src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -827,14 +841,13 @@ public Task WriteAsync(ReadOnlyMemory<byte> data, CancellationToken cancellation
             else
             {
                 HandleNonBodyResponseWrite();
-                return !firstWrite ? Task.CompletedTask : FlushAsync(cancellationToken);
+                return !firstWrite ? Task.CompletedTask : FlushAsyncInternal(cancellationToken);
             }
         }
 
-        public async Task WriteAsyncAwaited(Task initializeTask, ReadOnlyMemory<byte> data, CancellationToken cancellationToken)
+        public async Task InitializeAndWriteAsyncAwaited(Task initializeTask, ReadOnlyMemory<byte> data, CancellationToken cancellationToken)
```

Comment:
I don't like the "InitializeAnd" naming.  Ay Write or Flush call might result in initialization.

In this case, initialization has already begun by the time the method is called. The first argument being the initializeTask makes it pretty clear what's being awaited.